.actions/setup_tools.py

```diff
@@ -42,45 +45,92 @@ def _load_py_module(name: str, location: str) -> ModuleType:
     return py
 
 
+def _augment_requirement(ln: str, comment_char: str = "#", unfreeze: str = "all") -> str:
+    """Adjust the upper version contrains.
+
+    Args:
+        ln: raw line from requirement
+        comment_char: charter marking comment
+        unfreeze: Enum or "all"|"major"|""
```

Comment:
I don't quite understand the difference between these. Can you explain it?

Reply:
I think that the example help best to see the dirrence